ceph/ceph_admin/bootstrap.py

```diff
@@ -0,0 +1,91 @@
+"""Module that allows QE to interface with cephadm bootstrap CLI."""
+import logging
+from typing import Dict
+
+from ceph.ceph import ResourceNotFoundError
+from utility.utils import get_cephci_config
+
+from .common import config_dict_to_string
+from .typing_ import CephAdmProtocol
+
+logger = logging.getLogger(__name__)
+
+
+class BootstrapMixin:
+    """Add bootstrap support to the child class."""
+
+    def bootstrap(self: CephAdmProtocol, config: Dict) -> None:
+        """
+        Execute cephadm bootstrap with the passed kwargs on the installer node.
+
+        Bootstrap involves,
+          - Creates /etc/ceph directory with permissions
+          - CLI creation with bootstrap options with custom/default image
+          - Execution of bootstrap command
+
+        Args:
+            config: Key/value pairs passed from the test case.
+
+        Example:
+            config:
+                command: bootstrap
+                base_cmd_args:
+                    verbose: true
+                args:
+                    custom_image: true | false
+                    mon-ip: <node_name>
+                    mgr-id: <mgr_id>
+                    fsid: <id>
+        """
+        self.cluster.setup_ssh_keys()
+        self.set_tool_repo()
+        self.install()
+
+        cdn_cred = get_cephci_config().get("cdn_credentials")
+        cmd = "cephadm"
+
+        if config.get("base_cmd_args"):
+            cmd += config_dict_to_string(config["base_cmd_args"])
+
+        args = config.get("args")
+        custom_image = args.pop("custom_image", True)
+
+        if custom_image:
+            cmd += f" --image {self.config['container_image']}"
+
+        cmd += " bootstrap"
+        custom_image_args = (
+            " --registry-url registry.redhat.io"
+            " --registry-username {user}"
+            " --registry-password {password}"
+        )
+        cmd += custom_image_args.format(
+            user=cdn_cred.get("username"),
+            password=cdn_cred.get("password"),
+        )
+
+        # To be generic, the mon-ip contains the global node name. Here, we replace the
+        # name with the IP address. The replacement allows us to be inline with the
+        # CLI option.
+        mon_node = args.pop("mon-ip")
```

Comment:
Should we allow it be optional and take the `installer-node ip address` if the value is not provided?

Reply:
However currently node cannot be other than installer node.